core/src/main/java/io/substrait/type/proto/BaseProtoTypes.java

```diff
@@ -41,6 +48,13 @@ public BaseProtoTypes(Type.Nullability nullability) {
     INTERVAL_DAY = wrap(Type.IntervalDay.newBuilder().setNullability(nullability).build());
     INTERVAL_YEAR = wrap(Type.IntervalYear.newBuilder().setNullability(nullability).build());
     UUID = wrap(Type.UUID.newBuilder().setNullability(nullability).build());
+    VARCHAR = wrap(Type.VarChar.newBuilder().setNullability(nullability).build());
+    FIXED_CHAR = wrap(Type.FixedChar.newBuilder().setNullability(nullability).build());
+    FIXED_BINARY = wrap(Type.FixedBinary.newBuilder().setNullability(nullability).build());
+    DECIMAL = wrap(Type.Decimal.newBuilder().setNullability(nullability).build());
+    MAP = wrap(Type.Map.newBuilder().setNullability(nullability).build());
```

Comment:
Without a type you're going to need to get a builder, add the type, and then build it again.  Is it worth having a base type for this case?

Reply:
I think this pr can be close, this is not a good adding.

Reply:
VARCHAR, FIXED_CHAR, FIXED_BINARY all should be good to go.

DECIMAL, MAP, LIST, and STRUCT all have some sort of parameter and thus need to be functions.

Reply:
Hmm, except that those three types all expect to use the literal size as their parameter so without even a literal they aren't fully specified.  I agree, there aren't any more types that work with a simple type.

If we do want something for these they'd have to be functions (perhaps returning cached values) but there's not much perceivable gain there.

Reply:
> Hmm, except that those three types all expect to use the literal size as their parameter so without even a literal they aren't fully specified. I agree, there aren't any more types that work with a simple type.
> 
> If we do want something for these they'd have to be functions (perhaps returning cached values) but there's not much perceivable gain there.

Maybe VARCHAR, FIXED_CHAR, FIXED_BINARY also need a size parameter？

Reply:
I do wonder if these are going to be directly usable without a size parameter.  But I haven't written any code using these.  Let's see if someone else has an opinion.